File: src/main/java/requests/spark/websockets/objects/messages/outgoing/AnswerUpdate.java
package requests.spark.websockets.objects.messages.outgoing;

import game.actors.Answer;
import org.apache.log4j.Logger;
import requests.spark.websockets.objects.Message;
import requests.spark.websockets.objects.MessageType;

@MessageType("AnswerUpdate")
public class AnswerUpdate extends Message {
    private static Logger log = Logger.getLogger(AnswerUpdate.class);

    private Answer answer = null;

    public AnswerUpdate answer(Answer answer) {
        this.answer = answer;
        return this;
    }

    public void prepareToSend() {
        addResponseData("answerUUID", answer.getUuid());
        addResponseData("playerUUID", answer.getPlayer().getUuid());
        addResponseData("answerValue", answer.getAnswerValue());
    }
}

File: src/main/ngapp/src/app/wsObjects/answerUpdate.ts
import {Message} from "./message";

export class AnswerUpdate extends Message {

  private _answerUUID: string;
  private _playerUUID: string;
  private _answerValue: string;

  constructor() {
    super();
    this.type = "AnswerUpdate";
  }

  decodeResponse(msgRaw: any) {
    this._answerUUID = msgRaw.answerUUID;
    this._playerUUID = msgRaw.playerUUID;
    this._answerValue = msgRaw.answerValue;
  }

  get answerUUID(): string {
    return this._answerUUID;
  }

  get playerUUID(): string {
    return this._playerUUID;
  }

  get answerValue(): string {
    return this._answerValue;
  }
}
